zuul-core/src/main/java/com/netflix/zuul/message/Headers.java

```diff
@@ -548,19 +624,50 @@ private void truncate(int i) {
         }
     }
 
+    /**
+     * Checks if the given value is compliant with our RFC 7230 based check
+     */
+    private boolean isValid(String value) {
+        if (value == null || findInvalid(value) == ABSENT) {
+            return true;
+        }
+        Spectator.globalRegistry().counter("zuul.header.invalid.char").increment();
```

Comment:
You can store a reference to this counter in a static variable to avoid recalculating it each time.